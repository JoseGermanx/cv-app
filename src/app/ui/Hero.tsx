"use client";

import Link from "next/link";
import React from "react";
import Typed from "typed.js";
import Socialmedia from "./Socialmedia";


const Hero = () => {
  const el = React.useRef(null);

  React.useEffect(() => {
    const typed = new Typed(el.current, {
      strings: ["FullStack", "Frontend", "Backend"],
      typeSpeed: 100,
      loop: true,
      showCursor: true,
      backDelay: 1000,
    });

    return () => {
      // Destroy Typed instance during cleanup to stop animation
      typed.destroy();
    };
  }, []);

  React.useEffect(() => {
    const script = document.createElement('script');
    script.src = "//cdn.credly.com/assets/utilities/embed.js";
    script.async = true;
    document.body.appendChild(script);
  }, []);

  return (
    <div className="relative isolate px-6 pt-2 lg:px-8 h-auto mt-20 mb-10">
      <div
        className="absolute inset-x-0 -top-40 -z-20 transform-gpu overflow-hidden blur-3xl sm:-top-80"
        aria-hidden="true"
      >
        <div
          className="relative left-[calc(50%-11rem)] aspect-[1155/678] w-[36.125rem] -translate-x-1/2 rotate-[30deg] bg-gradient-to-tr from-[#a3bedd] to-[#403f4d] opacity-30 sm:left-[calc(50%-30rem)] sm:w-[72.1875rem]"
          style={{
            clipPath:
              "polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%)",
          }}
        />
      </div>
      <div className="mx-auto max-w-2xl py-4 sm:py-4 lg:py-4">
        <div className="hidden sm:mb-8 sm:flex sm:justify-center"></div>
        <div className="text-center">
          <h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl">
            <span className=" text-indigo-600">Software</span> Developer
          </h1>
          <h2 className="text-1xl font-bold tracking-tight text-gray-900 sm:text-2xl mt-3">
            <span ref={el} className=" text-gray-600"></span>
          </h2>
          <p className="mt-6 text-lg leading-8 text-gray-600">
            Hola, bienvenido a mi sitio web. Mi nombre es José Germán. Soy ingeniero, desarrollador de
            Software Full Stack e instructor de JavaScript.
          </p>
          <div className="mt-10 flex items-center justify-center gap-x-6">
            {/* <Link href="/projects" className="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600">
            Ver projectos
          </Link> */}
            <Link
              href="/cv_jose_martinez.pdf"
              className="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            >
              Descargar CV
            </Link>
            <Link
              href="https://techmarketingpulse.beehiiv.com/subscribe"
              target="_blank"
              className="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            >
              Suscríbete al newsletter
            </Link>
          </div>
        </div>
      </div>
      <Socialmedia />
    </div>
  );
};

export default Hero;
