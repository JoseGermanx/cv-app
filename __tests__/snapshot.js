import { render } from '@testing-library/react'
import Hero from "../src/app/ui/Hero"
 
it('renders homepage unchanged', () => {
  const { container } = render(<Hero />)
  expect(container).toMatchSnapshot()
})