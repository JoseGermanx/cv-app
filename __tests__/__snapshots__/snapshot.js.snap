// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="relative isolate px-6 pt-2 lg:px-8 h-auto mt-20 mb-10"
  >
    <div
      aria-hidden="true"
      class="absolute inset-x-0 -top-40 -z-20 transform-gpu overflow-hidden blur-3xl sm:-top-80"
    >
      <div
        class="relative left-[calc(50%-11rem)] aspect-[1155/678] w-[36.125rem] -translate-x-1/2 rotate-[30deg] bg-gradient-to-tr from-[#a3bedd] to-[#403f4d] opacity-30 sm:left-[calc(50%-30rem)] sm:w-[72.1875rem]"
        style="clip-path: polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%);"
      />
    </div>
    <div
      class="mx-auto max-w-2xl py-4 sm:py-4 lg:py-4"
    >
      <div
        class="hidden sm:mb-8 sm:flex sm:justify-center"
      />
      <div
        class="text-center"
      >
        <h1
          class="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl"
        >
          <span
            class=" text-indigo-600"
          >
            Software
          </span>
           Developer
        </h1>
        <h2
          class="text-1xl font-bold tracking-tight text-gray-900 sm:text-2xl mt-3"
        >
          <span
            class=" text-gray-600"
          />
          <span
            aria-hidden="true"
            class="typed-cursor"
          >
            |
          </span>
        </h2>
        <p
          class="mt-6 text-lg leading-8 text-gray-600"
        >
          Mi nombre es José Germán, tengo +10 años de experiencia como desarrollador web y actualmente ayudo a estudiantes de desarrollo Full Stack en JavaScript.
        </p>
        <div
          class="mt-10 flex items-center justify-center gap-x-6"
        >
          <a
            class="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            href="/cv_jose_martinez.pdf"
          >
            Descargar CV
          </a>
          <a
            class="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            href="https://www.linkedin.com/newsletters/7078460407316635648/"
            target="_blank"
          >
            Suscríbete al newsletter
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex-col mt-9"
      >
        <div
          class="flex justify-center"
        >
          <div
            class="group relative"
          >
            <div
              class="aspect-h-1 aspect-w-1 w-full overflow-hidden group-hover:opacity-75 lg:h-auto mx-4"
            >
              <img
                alt="GitHub"
                class=" h-5 w-5 object-cover object-center lg:h-10 lg:w-10"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="https://simpleicons.org/icons/github.svg"
                style="color: transparent;"
                width="40"
              />
              <a
                href="https://www.github.com/josegermanx/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                  class="absolute inset-0"
                />
              </a>
            </div>
          </div>
          <div
            class="group relative"
          >
            <div
              class="aspect-h-1 aspect-w-1 w-full overflow-hidden group-hover:opacity-75 lg:h-auto mx-4"
            >
              <img
                alt="Web"
                class=" h-5 w-5 object-cover object-center lg:h-10 lg:w-10"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="https://simpleicons.org/icons/googlechrome.svg"
                style="color: transparent;"
                width="40"
              />
              <a
                href="https://www.jgxdev.com/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                  class="absolute inset-0"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
